### ISSUE TITLE
Provide a method to get the pager's element ID.

### ISSUE THREAD

**Issue by argiepiano** [backdrop/backdrop-issues#5958]
## Description of the need

This is a cross-port from https://www.drupal.org/node/702778, and part of the task list #5770. The D7 issue added a method, `getElement()`, to `PagerDefault` that returns the element ID of the pager. Having this method allows developers an easier way to have more than one pager in the same page without conflicts. 

Before this new method, the only way to handle more than one pager in the same page was to hard-code the element ID through `PagerDefault::element()`. This doesn't make much sense and can lead to unexpected conflicts if two different modules need to add a pager to the same page. `PagerDefault` already has a way to avoid those conflicts by keeping track of IDs in a static variable, but until this feature was added in D7, it was impossible to get the internally assigned element ID (which is a protected property). 

PR coming.

**Comment by argiepiano** [comment]
PR https://github.com/backdrop/backdrop-issues/issues/5958

The PR is simple - it adds a method that returns the value of the protected property `PagerDefault::$element`. In addition it modifies `testElementNumbers()`. 

It's a direct port from D7.

I have somehow lost the ability to edit the OP of #5770, and therefore I can't add a reference to this issue over there. Can someone edit that? The D7 id is 702778.  While you are there, would someone please restore my permission to edit posts in the issue queue?  I used to be able to do that.

**Comment by argiepiano** [comment]
There is  a coding standard failure - but it indicates a line this PR did not modify (?).

To test the PR, you can use this little custom module:
- Be user to use `devel_generate` to generate a bunch of nodes and users. 
- Enable the module and visit `/admin/testing-stuff`

This page will show a table of nodes and a table of users, each with its own functional pager. You need more than 3 nodes and 3 users for the pager to show up. 

[testing_stuff.zip](https://github.com/backdrop/backdrop-issues/files/10605930/testing_stuff.zip)

**Comment by klonos** [comment]
> There is a coding standard failure - but it indicates a line this PR did not modify (?).

It the `@return` line in the docblock of `PagerDefault::getElement()`, which this PR introduces, so it needs to be fixed. Simply adding the type returned by it (`int`) should stop PHPCS from complaining:

Tested on my local (with the PR merged) vs. a vanilla 1.24.0 demo site, and it works as expected:
![image](https://user-images.githubusercontent.com/2423362/216782846-54dea72f-51bd-4647-a6eb-a2e6c3dbee1c.png)

So WFM, but NW pending on fixing that PHPCS failure (otherwise code looks good and matches what was merged in D7).

**Comment by argiepiano** [comment]
Thanks @klonos! I missed that. It's fixed now.

**Comment by klonos** [comment]
Perfect 👍🏼 ...RTBC now.

**Comment by BWPanda** [comment]
Maybe I don't understand this issue properly, but I tested this on my sandbox and it doesn't seem to work...

I Devel Generated a bunch of nodes (both Pages and Posts). I then created a view with two block displays - one showing Page nodes and one showing Post nodes. Both displays were set to have a pager. I added both blocks to the footer of my Home layout and sure enough I get two blocks showing different content, each with their own pager.

Without applying the patch from this PR, I click on page 2 of the first block and the page reloads, showing page two of both blocks:

![image](https://user-images.githubusercontent.com/2385329/221148126-15d716fa-f786-4288-8b84-f79c17eacde5.png)

So I assume that's the problem we're trying to solve here. However even after applying the PR's patch, nothing changes...

But then I don't see how it would change anything, since it just adds a new function but doesn't call it from anywhere. Is there another issue to go along with this one that uses this new function? Or am I totally missing the point here?

**Comment by klonos** [comment]
> ...it just adds a new function but doesn't call it from anywhere.

@BWPanda please read the issue summary for details. This bit specifically:
> Having this method allows developers an easier way to have more than one pager in the same page without conflicts.

@argiepiano provided a test module in https://github.com/backdrop/backdrop-issues/issues/5958#issuecomment-1416653858 that when installed in a vanilla Backdrop it demonstrates the problem. Installing the same module on your local (with the patch applied) or in the PR sandbox should work as expected.

You are right in that we aren't using this in core, and your steps actually demonstrate a use case in core that is not fixed. The question is should we fix that as part of this issue/PR here (which is a backport of what went into Drupal core 7.92), or a separate issue specifically for Views? (I'm assuming you have used Views for the blocks, am I right?)

**Comment by BWPanda** [comment]
Thanks @klonos. I guess I just get a bit confused when we add a new feature to Backdrop without also utilising it anywhere, or adding code that allows fixing bugs without actually fixing the bugs themselves...

> The question is should we fix that as part of this issue/PR here [...], or a separate issue specifically for Views?

Either/or. But I think fixing it in this issue has more value since then we can see actual benefit to it, rather than just an example of what's possible in contrib/custom code.

> I'm assuming you have used Views for the blocks, am I right?

Correct.

**Comment by argiepiano** [comment]
This issue is part of a task list for D7 backports. I would suggest that we open another issue for the problem @BWPanda encountered, and leave this as is. I haven't looked closely at what @BWPanda reported, but my guess is that, before this patch, fixing that Views issue was not possible (or was more difficult). So, while the feature here doesn't explicitly fix a bug, it provides tools to fix other bugs like the one reported above.

**Comment by klonos** [comment]
> This issue is part of a task list for D7 backports. I would suggest that we open another issue for the problem @BWPanda encountered, and leave this as is.

I think the same 👍🏼

**Comment by quicksketch** [comment]
Merged into 1.x and 1.24.x. Thanks @argiepiano, @BWPanda, and @klonos!
